dig/dig.go

```diff
@@ -62,15 +89,23 @@ func (g *Graph) Register(c interface{}) error {
 
 	switch ctype.Kind() {
 	case reflect.Func:
-		if ctype.NumOut() != 1 {
+		if ctype.NumOut() == 0 || ctype.NumOut() > 2 {
```

Comment:
I meant to add something similar to "second return value of constructor must be error"